### PR TITLE
fix: Sever side error

### PR DESCRIPTION
**Pull request by sumit-158**
### What
- Remove expectation from [main.py](https://github.com/openfoodfacts/facets-knowledge-panels/blob/9b1c3562c67ac0fe40a7b017386ce6c592f95f9c/app/main.py#L115) and putting deep in the code 

### Part of 
- #85

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -109,11 +109,9 @@ async def knowledge_panel(
...
             # but we don't want to sacrifice whole result for a single failure
             # as most panels depends on external resources that may not be available
-            try:
-                if soon_value.value:
-                    panels.update(soon_value.value)
-            except Exception as e:
-                logging.exception(msg=e)
+
+            if soon_value.value:
+                panels.update(soon_value.value)
```
Comment (alexgarel) [inline app/main.py:114]: Did you investigated why this try/except did not get exceptions ?
I'm a bit reluctant to loose this pattern for it was more robust and clean to have it here.

Comment (sumit-158) [reply app/main.py:114]: It seems "task_group.soonify" doesn't catch the exceptions.

Comment (alexgarel) [reply app/main.py:114]: according to the docs, it should trigger it as we ask the value, but maybe we are doing something wrong (maybe we should stay in the "with" ?

I would try to simply indent the code to keep it under "with". That would be the only change you have to do to the whole code.

Comment (alexgarel) [reply app/main.py:114]: @sumit-158 I will try to look into this next week. It intrigues me !

Comment (alexgarel) [reply app/main.py:114]: You are right https://github.com/tiangolo/asyncer/blob/main/asyncer/_main.py#L153

That's a bit sad !

So maybe we should add a decorator around our functions to catch any exception.

```
@no_fail
async def data_quality_kp(self, ...
```

Do you want to try to write such an async decorator ?

Comment (sumit-158) [reply app/main.py]: @alexgarel made the function, please do check it :-)

File: app/exception_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import functools
+import logging
+
+
+def exceptionCatcher():
```
Comment (alexgarel) [inline app/exception_wrapper.py:5]: It's a function it should use snake_case. More over this is a decorator, so you'd better call it with a name that feels natural when place above the function name.

What about no_fail or no_exception ?

File: app/exception_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        async def wrapper(*args, **kwargs):
+            try:
+                return await func(*args, **kwargs)
+            except Exception:
+                issue = "exception in " + func.__name__ + "\n"
+                logging.exception(issue)
+
+        return wrapper
+
+    return decorator
```
Comment (alexgarel) [inline app/exception_wrapper.py:25]: kudos for implementing your first decorator :muscle:

**Review by alexgarel — APPROVED** [review]
Great job @sumit-158 !

Can we at least change the case before merging ?

File: app/exception_wrapper.py (hunk 2 above)
Comment (sumit-158) [reply app/exception_wrapper.py]: I like "no_exception" :-)

File: app/exception_wrapper.py (hunk 3 above)
Comment (sumit-158) [reply app/exception_wrapper.py:25]: Thank you :-)

**Pull request merged**